modules/indexer/issues/indexer.go

```diff
@@ -347,5 +348,9 @@ func SearchIssuesByKeyword(repoIDs []int64, keyword string) ([]int64, error) {
 	for _, r := range res.Hits {
 		issueIDs = append(issueIDs, r.ID)
 	}
+	i, err := strconv.ParseInt(keyword, 10, 64)
+	if err == nil {
+		issueIDs = append(issueIDs, i)
```

Comment:
I think there are some problems:

1. `i` is not suitable for a formal variable name
2. Do you mean `issueID` or `issueIndex`?
3. If we do not use `repoIDs` to filter issues, the unrelated issues from other repositories would appear in result.

Reply:
1. agreed. Which name would be better? `keywordAsInt`?
2. what is `issueIndex`? I want to add an ID to the found IDs
3. My use-case is for the repo search, so the repository is known and it does not show issues from any other repo. Do you think the API could return something wrong?
`r.ID` from line 348 is just a issue number too and not the database ID.

Reply:
`2` If you see http://domain/org/repo/issues/123, the 123 is the issue index. There can also be another issue with index 123 in another repo: http://domain/org/repo-other/issues/123 .  Issue ID is the unique internal auto-increment database column for each issue, users seldom see the issue IDs, instead, users use issue index everywhere. Of course, Gitea use issue ID internally everywhere.

`3` I believe the `r.ID` is the unique issue ID, not issue index. However, if there is another query filter outside `SearchIssuesByKeyword` to filter the issue IDs with repositories again, the final result can be correct, but it is still not safe to return a non-related issue from SearchIssuesByKeyword.

Reply:
In my mine: an easy way to find issues by the indexes can be `put the issue index into the search indexex`, then we do not need to change SearchIssuesByKeyword, then the issue indexes can be found correctly.

Reply:
oh I see, it worked only because of my fresh database where issueIndex was also issueID :/

Reply:
>put the issue indexes into the search indexes (make the issue indexes searchable)

I am not familiar enough with the code base to know how to do this

Reply:
You can take a look at `IndexerData`

```go
// IndexerData data stored in the issue indexer
type IndexerData struct {
	ID       int64    `json:"id"`
	RepoID   int64    `json:"repo_id"`
	Title    string   `json:"title"`
	Content  string   `json:"content"`
	Comments []string `json:"comments"`
	IsDelete bool     `json:"is_delete"`
	IDs      []int64  `json:"ids"`
}
```

Follow the way how `Content` are put into search indexer.

Reply:
I feel like I am close, yet it won't work. What am I missing?

Reply:
Have you refreshed your search indexer? Maybe you can dump the document in the search engine to see whether they are correct.